lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -477,7 +481,7 @@ def create_or_update_target_group(connection, module):
                 module.fail_json_aws(e, msg="Couldn't update target group")
 
         # Do we need to modify targets?
-        if module.params.get("modify_targets"):
+        if module.params.get("purge_targets"):
```

Comment:
This change will mean that new targets will not get registered if `purge_targets` is False, which is not what would be expected.

We could really use a test suite to catch this kind of thing.

Reply:
Yeah, maybe purge_targets is not the right name for this. @willthames Do you have a problem with this fix if the option name remains 'modify_targets'?

Reply:
This seems like a problem with overloading what purge_ means. The code does what I'd expect, but not what I'd expect a purge_targets option to do. @Daemoen, what do you think?